src/limbo/model/gp.hpp

```diff
@@ -117,11 +119,14 @@ namespace limbo {
             void add_sample(const Eigen::VectorXd& sample, const Eigen::VectorXd& observation, double noise)
             {
                 if (_samples.empty()) {
+		  if (_dim_in != sample.size()) {
                     _dim_in = sample.size();
                     _kernel_function = KernelFunction(_dim_in); // the cost of building a functor should be relatively low
-
-                    _dim_out = observation.size();
-                    _mean_function = MeanFunction(_dim_out); // the cost of building a functor should be relatively low
+		  }
+		  if (_dim_out != observation.size()) {
```

Comment:
I double checked with my experiences and as the CI passes, I think it does not break anything.